### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by elyutikov**

### CODE REVIEW

File: new/src/main/scala/io/github/weakteam/model/auth/package.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.github.weakteam.model
+
+import eu.timepit.refined.types.string.NonEmptyString
+import io.estatico.newtype.macros.newtype
+
+package object auth {
```
Comment (Ssstlis) [inline new/src/main/scala/io/github/weakteam/model/auth/package.scala:6]: I think this things may be placed in usual `object` named `newtypes`

File: new/src/main/scala/io/github/weakteam/model/auth/AuthenticationResult.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.github.weakteam.model.auth
+
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto._
+import io.circe.refined._
+
+final case class AuthenticationResult(accessToken: AccessToken, refreshToken: Option[RefreshToken])
+
+object AuthenticationResult {
+  implicit val brandDecoder: Decoder[AuthenticationResult] = deriveDecoder[AuthenticationResult]
```
Comment (Ssstlis) [inline new/src/main/scala/io/github/weakteam/model/auth/AuthenticationResult.scala:10]: Why you didn't use the `@JsonCodec` macro annotation?

File: new/src/main/scala/io/github/weakteam/model/package.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package io.github.weakteam
+
+import io.circe.{Decoder, Encoder}
+import io.estatico.newtype.Coercible
+import io.estatico.newtype.ops._
+
+package object model {
+
+  implicit def coercibleDecoder[A: Coercible[B, *], B: Decoder]: Decoder[A] =
```
Comment (Ssstlis) [inline new/src/main/scala/io/github/weakteam/model/package.scala:9]: Is it used somewhere? I didn't notice

Comment (elyutikov) [reply new/src/main/scala/io/github/weakteam/model/package.scala:9]: it is for newtype codecs

**Comment by Ssstlis** [comment]
Ca you please add some checklist for status of your PR to topic post?)

**Pull request closed without merging**